backend/corpora/common/entities/dataset.py

```diff
@@ -148,28 +145,25 @@ def dataset_and_asset_deletion(self):
         self.tombstone_dataset_and_delete_child_objects()
 
     @staticmethod
-    def new_processing_status():
+    def new_processing_status() -> dict:
         return {
             "upload_status": UploadStatus.WAITING,
             "upload_progress": 0,
             "processing_status": ProcessingStatus.PENDING,
         }
 
-    def copy_csv_to_s3(self, csv_file):
-        s3 = boto3.resource("s3", endpoint_url=os.getenv("BOTO_ENDPOINT_URL"))
-        cellxgene_bucket = os.getenv("CELLXGENE_BUCKET", f"hosted-cellxgene-{os.environ['DEPLOYMENT_STAGE']}")
-        s3.meta.client.upload_file(csv_file, cellxgene_bucket, csv_file)
+    def copy_csv_to_s3(self, csv_file: str) -> str:
+        s3_file = f"{get_cxg_bucket_path(self.deployment_directories[0])}/genesets.csv"
+        cxg_bucket.upload_file(csv_file, s3_file)
+        return s3_file
 
-    def generate_tidy_csv_for_all_linked_genesets(self, csv_file=None):
-        if csv_file is None:
-            base, ext = os.path.splitext(self.deployment_directories[0].url)
-            if ext is not None:  # strip extension, if any
-                csv_file = f"{base}-genesets.csv"
+    def generate_tidy_csv_for_all_linked_genesets(self, csv_file_path: str) -> str:
+        csv_file = os.path.join(csv_file_path, "geneset.csv")
```

Comment:
hardcoding geneset.csv name, and combining it with the file path passed in as the csv_file_path parameter.

Reply:
what is the reason for this change?